oauth2/oauth2_auth_code_test.go

```diff
@@ -1033,12 +1034,63 @@ func TestAuthCodeWithMockStrategy(t *testing.T) {
 						hs := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 							assert.Equal(t, r.Header.Get("Content-Type"), "application/json; charset=UTF-8")
 
+							expectedGrantedScopes := []string{"openid", "offline", "hydra.*"}
+							expectedSubject := "foo"
+
 							var hookReq hydraoauth2.RefreshTokenHookRequest
 							require.NoError(t, json.NewDecoder(r.Body).Decode(&hookReq))
-							require.Equal(t, hookReq.Subject, "foo")
-							require.ElementsMatch(t, hookReq.GrantedScopes, []string{"openid", "offline", "hydra.*"})
+							require.Equal(t, hookReq.Subject, expectedSubject)
+							require.ElementsMatch(t, hookReq.GrantedScopes, expectedGrantedScopes)
 							require.ElementsMatch(t, hookReq.GrantedAudience, []string{})
 							require.Equal(t, hookReq.ClientID, oauthConfig.ClientID)
+							require.NotEmpty(t, hookReq.Session)
+							require.Equal(t, hookReq.Session.Subject, expectedSubject)
+							require.Equal(t, hookReq.Session.ClientID, oauthConfig.ClientID)
+							require.Equal(t, hookReq.Session.Extra, map[string]interface{}{})
+							require.NotEmpty(t, hookReq.Requester)
+							require.Equal(t, hookReq.Requester.ClientID, oauthConfig.ClientID)
+							require.ElementsMatch(t, hookReq.Requester.GrantedScopes, expectedGrantedScopes)
+
+							assertx.EqualAsJSONExcept(t, hookReq.Session, json.RawMessage(`{
+								"allowed_top_level_claims": [],
+								"client_id": "app-client",
+								"consent_challenge": "",
+								"exclude_not_before_claim": false,
+								"extra": {},
+								"id_token": {
+									"headers": {
+										"extra": {"kid":"public:hydra.openid.id-token"}
+									},
+									"id_token_claims": {
+										"acr": "1",
+										"amr": null,
+										"at_hash": "",
+										"c_hash": "",
+										"aud": ["app-client"],
+										"ext": {"sid":""},
+										"iss": "http://localhost:4444/",
+										"jti": "",
+										"nonce": "",
+										"sub": "foo"
+									},
+									"subject": "foo",
+									"username": ""
+								}
+							}`), []string{
+								"kid",
```

Comment:
Could not get this one consistent in the test, sometimes it is empty, sometimes is not.

Reply:
Ah yes, the kid is randomly generated when using the JSON Web Token strategy, and empty when using the opaque strategy. This is ok!